types/basetypes/list_type.go

```diff
@@ -65,6 +65,20 @@ func (l ListType) ValueFromTerraform(ctx context.Context, in tftypes.Value) (att
 	if in.Type() == nil {
 		return NewListNull(l.ElementType()), nil
 	}
+
+	// MAINTAINER NOTE:
```

Comment:
Here is my "brain dump" attempt at describing how we could support dynamics inside of `ListType`. Please let me know if this makes sense 😆